cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -701,51 +706,62 @@ def _validate_column_dependencies(
         self, df: pd.DataFrame, df_name: str, column_name: str, dependencies: List[dict]
     ) -> pd.Series:
         """
-        Validates subset of columns based on dependencies, for instance development_stage_ontology_term_id has
-        dependencies with organism_ontology_term_id -- the allowed values depend on whether organism is human, mouse
-        or something else.
+        Validates subset of columns based on dependencies, e.g., if a column like 'development_stage_ontology_term_id'
+        has allowed values that depend on other column values, either in the same dataframe (as specified by the
+        keyword "column") or in the uns dictionary (as specified by the keyword "uns_key").
 
-        After performing all validations, it will return the column that has been stripped of the already validated
-        fields -- this has to still be validated.
-
-        :param pd.DataFrame df: pandas dataframe containing the column to be validated
-        :param str df_name: the name of dataframe in the adata object, e.g. "obs"
-        :param str column_name: the name of the column to be validated
-        :param list dependencies: a list of dependency definitions, which is a list of column definitions with a "rule"
+        Returns a Series containing values from the column that were not matched by any rule.
         """
-
         all_rules = []
+
         for dependency_def in dependencies:
             terms_to_match = set()
-            column_to_match = dependency_def["rule"]["column"]
-            if "match_ancestors_inclusive" in dependency_def["rule"]:
-                ancestors = dependency_def["rule"]["match_ancestors_inclusive"]["ancestors"]
+            rule = dependency_def["rule"]
+            column_to_match = rule.get("column")
+            uns_key_to_match = rule.get("uns_key")
+
+            # Build the set of terms to match
+            if "match_ancestors_inclusive" in rule:
+                ancestors = rule["match_ancestors_inclusive"]["ancestors"]
                 for ancestor in ancestors:
-                    terms_to_match.update(get_descendants(ONTOLOGY_PARSER, ancestor, True))
-            if "match_exact" in dependency_def["rule"]:
-                terms_to_match.update(dependency_def["rule"]["match_exact"]["terms"])
+                    terms_to_match.update(get_descendants(ONTOLOGY_PARSER, ancestor, include_self=True))
+            if "match_exact" in rule:
+                terms_to_match.update(rule["match_exact"]["terms"])
+
             try:
-                match_query = df[column_to_match].isin(terms_to_match)
+                if column_to_match:
+                    match_query = df[column_to_match].isin(terms_to_match)
+                elif uns_key_to_match:
```

Comment:
`_validate_column_dependencies` was extended to support `uns_key` for fields like `development_stage_ontology_term_id` that depend on `uns["organism_ontology_term_id"]`